csrc/ir_utils.h

```diff
@@ -429,5 +429,28 @@ void validateDomainEquivalence(
     const std::vector<IterDomain*>& initial_domain,
     const std::vector<IterDomain*>& derived_domain);
 
+//! Compute the IterType of an IterDomain that has been resized. If the output
+//! is size 1, or the output uses no input elements, this function returns
+//! Broadcast. Otherwise, it returns Iteration.
+inline IterType resizeOutputIterType(
+    int64_t in_extent,
+    int64_t out_extent,
+    int64_t left,
+    int64_t right) {
+  TORCH_CHECK(out_extent >= 0, "Resized extent must be non-negative.");
+  if (
+      // negative padding sums to input extent. Output is zero-dimensional
+      out_extent == 0 ||
+      // input overlaps output
+      left + in_extent > 0 || right + in_extent > 0) {
+    return IterType::Iteration;
```

Comment:
For example, when `in_extent: 2, left: 1, right: -2`, then `out_extent` is 1. Isn't it a broadcast?

Reply:
You're right. This was a hold-over from when I had planned to replace pure-pad resizes with `full()`, in which case the result would be like an `expand()`. I just pushed a change that removes this function altogether as the logic is now simply "`Broadcast` if `out_extent==1` else `Iteration`".